routes/routes.js

```diff
@@ -55,24 +54,12 @@ var appRouter = function (app) {
     app.post("/deploy", async function (req, res) {
         let tokenId = req.body.tokenId;
         try {
-            await buildUtil.makeBuildDir(tokenId);
-            await buildUtil.prepContractDir(tokenId);
-            await buildUtil.buildCrowdsaleContract(tokenId);
-
-            let tokenObject = await tokenService.getTokenByTokenId(tokenId);
-            let tokenJSON = tokenObject.toJSON();
-
-            let crowdsaleObject = await crowdsaleService.getCrowdsaleByTokenId(tokenId);
-            let crowdsaleJSON = crowdsaleObject.toJSON();
-
-            await buildUtil.buildTokenContract(tokenId, tokenJSON.name, tokenJSON.symbol);
-
-            await buildUtil.prepMergeDir(tokenId);
-            await buildUtil.mergeCrowdsaleContract(tokenId);
-            let contractAddress = await buildUtil.deployCrowdsaleContract(tokenId, crowdsaleJSON);
-
-            log.info('deployed address:', contractAddress);
-            res.send(contractAddress);
+            if (!tokenId) {
+                res.status(422).send('missing tokenId');
+                return;
+            }
+            let result = await deployService.registerDeployment(req.body);
```

Comment:
deployJobService is job Q, I recommend upstream(e.g. the /deploy request) always talk with the job Q. 

So the workflow should be:
1) JobQ start to loop at the beginning
2) Insert parameters in to JobQ when getting request
3) JobQ start a deployment job thread with parameter

Otherwise, it's confusing. As the benefit is obvious: 
1) deployment job is transparent to upstream
2) single responsibility
3) easier for testing

Reply:
updated